pkg/install/drbd/drbd_adapter.go

```diff
@@ -98,375 +104,537 @@ func HandelDRBDConfigs(clusterInstance *hwameistoriov1alpha1.Cluster) {
-		clusterInstance.Spec.DRBD.DeployOnMaster = defaultDeployOnMaster
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return adapterCreatedJobNum, fmt.Errorf("failed to new clientset: %v", err)
 	}
-
-	if clusterInstance.Spec.DRBD.Shipper == nil {
-		clusterInstance.Spec.DRBD.Shipper = &operatorv1alpha1.ImageSpec{}
+	nodeList, err := clientset.CoreV1().Nodes().List(context.TODO(), v1.ListOptions{})
+	if err != nil {
+		return adapterCreatedJobNum, err
 	}
-	if clusterInstance.Spec.DRBD.Shipper.Registry == "" {
-		clusterInstance.Spec.DRBD.Shipper.Registry = defaultImageRegistry
+	jobStatus := make(map[string]bool)
+	var wg sync.WaitGroup
+	for _, node := range nodeList.Items {
+		wg.Add(1)
+		go monitorNodeJobs(cli, clientset, node, jobStatus, &wg)
 	}
-	if clusterInstance.Spec.DRBD.Shipper.Repository == "" {
-		clusterInstance.Spec.DRBD.Shipper.Repository = defaultShipperRepository
+	wg.Wait()
+	allSucceeded := true
+	for _, status := range jobStatus {
+		if !status {
+			allSucceeded = false
+			break
+		}
 	}
 
-	if clusterInstance.Spec.DRBD.Shipper.Tag == "" {
-		clusterInstance.Spec.DRBD.Shipper.Tag = defaultShipperChar
+	if allSucceeded {
+		return adapterCreatedJobNum, nil
+	} else {
+		err = fmt.Errorf("Some jobs failed.")
+		return adapterCreatedJobNum, err
 	}
+}
 
-	if clusterInstance.Spec.DRBD.DRBDVersion == "" {
-		clusterInstance.Spec.DRBD.DRBDVersion = defaultDRBDVersion
-	}
-	if clusterInstance.Spec.DRBD.Upgrade == "" {
-		clusterInstance.Spec.DRBD.Upgrade = defaultDRBDUpgrade
-	}
-	if clusterInstance.Spec.DRBD.CheckHostName == "" {
-		clusterInstance.Spec.DRBD.CheckHostName = defaultCheckHostName
-	}
-	if clusterInstance.Spec.DRBD.UseAffinity == "" {
-		clusterInstance.Spec.DRBD.UseAffinity = defaultUseAffinity
-	}
-	if clusterInstance.Spec.DRBD.NodeAffinity == nil {
-		clusterInstance.Spec.DRBD.NodeAffinity = &corev1.NodeAffinity{}
+func monitorNodeJobs(cli client.Client, clientset *kubernetes.Clientset, node corev1.Node, jobStatus map[string]bool, wg *sync.WaitGroup) {
+	defer wg.Done()
+	jobName := "drbd-adapter-" + node.Name
+	for {
+		// Check the current status of the job
+		job, err := clientset.BatchV1().Jobs(namespace).Get(context.TODO(), jobName, v1.GetOptions{})
+		if err == nil {
+			if isJobComplete(job) {
+				jobStatus[node.Name] = true
+				break
+			} else if isJobFailed(job) {
+				// Wait for the job to be deleted manually
+				if waitForJobDeletion(clientset, jobName) {
+					// After deletion, create the job again
+					CreateDRBDJob(cli, node)
+					if watchJob(clientset, jobName) {
+						jobStatus[node.Name] = true
+						break
+					}
+
+				}
+			} else {
+				// The job is still running, continue watching
+				if watchJob(clientset, jobName) {
+					jobStatus[node.Name] = true
+					break
+				}
+			}
+		} else {
+			// Job does not exist, create it
+			CreateDRBDJob(cli, node)
+			if watchJob(clientset, jobName) {
+				jobStatus[node.Name] = true
+				break
+			}
+		}
 	}
```

Comment:
separate this func as 2 funcs: `monitorDRBDSetupJob` and `setupDRBDOnNode | CreateDRBDJob` may be better